Internal Prettier stopped working (PHP)

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Sometime in the last few Zed Preview updates, the internal prettier instance stopped formatting php code for me. The code is still getting formatted - I assume by the intelephense LSP - but it suddenly stopped following the prettier config I have set up in my zed settings and that led me down a rabbit hole which led me to conclude that prettier itself is failing.

Here's the prettier config that I mentioned from my zed settings.json:

````
  "prettier": {
    "overrides": [
      {
        "files": [
          "*.php",
          "*.phtml"
        ],
        "options": {
          "phpVersion": "8.0",
          "braceStyle": "1tbs",
          "tabWidth": 2
        }
      }
    ]
  },
````

Up until recently, that config has been working fine. Yesterday, I tweaked some code, saved, and suddenly the whole file had a bunch of changes where opening braces were moved to the next line ("1tbs" keeps them on the same line).

I created a simple test project with a simple .php file and requested a format and I got a formatting failed message in the zed footer and when I clicked to see the logs, here's what it says:

````
2024-07-05T12:29:26-05:00 [ERROR] default prettier instance failed to format buffer

Caused by:
    0: prettier format request
    1: error during message '{"jsonrpc":"2.0","id":3,"method":"prettier/format","params":{"text":"..snip..","options":{"plugins":[],"parser":null,"filepath":"/Users/pjv/development/projects/test/test.php","prettierOptions":{"useTabs":false,"tabWidth":2,"printWidth":80}}}}' handling: UndefinedParserError: No parser could be inferred for file "/Users/pjv/development/projects/test/test.php".
````

Could this be related to #13666 and/or #13660 ?

### Environment

Zed: v0.143.2 (Zed Preview)
OS: macOS 14.5.0
Memory: 64 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/assets/settings/default.json
+++ b/assets/settings/default.json
@@ -775,7 +775,8 @@
     "PHP": {
       "prettier": {
         "allowed": true,
-        "plugins": ["@prettier/plugin-php"]
+        "plugins": ["@prettier/plugin-php"],
+        "parser": "php"
       }
     },
     "Ruby": {

